modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationReply.java

```diff
@@ -93,44 +121,80 @@ private void generateBody() throws WindowsOperationException {
     private SyncmlBody generateStatuses() {
         SyncmlBody sourceSyncmlBody = syncmlDocument.getBody();
         SyncmlHeader sourceHeader = syncmlDocument.getHeader();
+        Status headerStatus = null;
         SyncmlBody syncmlBodyReply = new SyncmlBody();
         List<Status> status = new ArrayList<Status>();
-        Status headerStatus =
-                new Status(HEADER_COMMAND_REFERENCE_ID, sourceHeader.getMsgID(), HEADER_STATUS_ID,
-                        HEADER_COMMAND_TEXT, sourceHeader.getSource().getLocURI(),
-                        String.valueOf(Constants.SyncMLResponseCodes.AUTHENTICATION_ACCEPTED));
-        status.add(headerStatus);
+        List<Status> sourceStatus = sourceSyncmlBody.getStatus();
+        if (sourceStatus.size() == 0) {
+            headerStatus =
+                    new Status(HEADER_COMMAND_ID, sourceHeader.getMsgID(), HEADER_STATUS_ID,
+                            HEADER_COMMAND_TEXT, sourceHeader.getSource().getLocURI(),
+                            String.valueOf(Constants.SyncMLResponseCodes.AUTHENTICATION_ACCEPTED));
+            status.add(headerStatus);
+        } else {
+
+            for (int i = 0; i < sourceStatus.size(); i++) {
+                Status st = sourceStatus.get(i);
+                if (st.getChallenge() != null && HEADER_COMMAND_TEXT.equals(st.getCommand())) {
+
+                    headerStatus =
+                            new Status(HEADER_COMMAND_ID, sourceHeader.getMsgID(), HEADER_STATUS_ID,
+                                    HEADER_COMMAND_TEXT, sourceHeader.getSource().getLocURI(),
+                                    String.valueOf(Constants.SyncMLResponseCodes.AUTHENTICATION_ACCEPTED));
+                    status.add(headerStatus);
+                }
+            }
+        }
+        if (sourceSyncmlBody.getResults() != null) {
+            int ResultCommandId = ++HEADER_COMMAND_ID;
```

Comment:
A better option would be to use the constant "HEADER_COMMAND_ID" as the default value and use a separate variable to keep track of id counts, etc.